### ISSUE TITLE
Spotify Login Fails

### ISSUE THREAD

**Issue by TomCasavant** [music-assistant/hass-music-assistant#289]
When initially configuring the integration, I'll get this message:
`Retrying setup: Setup failed of provider spotify: Login failed for user (my username)`
And it never succeeds to login. *Unless* I reboot the homeassistant and then it usually (except for this most recent reboot) manages to login successfully

Looking at the Homeassistant Core Logs, I have these:
`Login failed for user (my username)`
and
```
First occurred: 2:44:29 PM (13 occurrences)
Last logged: 2:59:36 PM
Setup failed of provider spotify: Login failed for user (username)
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/music_assistant/controllers/music/__init__.py", line 369, in _register_provider
    if await provider.setup():
  File "/usr/local/lib/python3.9/site-packages/music_assistant/controllers/music/providers/spotify.py", line 72, in setup
    raise LoginFailed(f"Login failed for user {self.config.username}")
music_assistant.models.errors.LoginFailed: Login failed for user (username)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/mass/__init__.py", line 105, in async_setup_entry
    await mass.setup()
  File "/usr/local/lib/python3.9/site-packages/music_assistant/mass.py", line 83, in setup
    await self.music.setup()
  File "/usr/local/lib/python3.9/site-packages/music_assistant/controllers/music/__init__.py", line 61, in setup
    await self._register_provider(prov_cls(self.mass, prov_conf), prov_conf)
  File "/usr/local/lib/python3.9/site-packages/music_assistant/controllers/music/__init__.py", line 372, in _register_provider
    raise SetupFailedError(
music_assistant.models.errors.SetupFailedError: Setup failed of provider spotify: Login failed for user (username)
```

Looking through some of the most recent issues, it looks like #230 was experiencing similar issues (though mine doesn't seem to be impacting homeassistant startup times, I have a 19s startup time for Music Assistant). I'm not sure if anything can be done w/ this, but it's something I can probably live with if necessary

**Comment by marcelveldt** [comment]
This sounds to me like a race condition where network connection is not yet available at startup time of Music Assistant.
I'll put a small guard in the code to retry the login twice.

**Comment by marcelveldt** [comment]
should be fixed in latest version (2022.6 beta 2) - if not please reopen. Thanks!
